R/controls.R

```diff
@@ -0,0 +1,38 @@
+#' @param html the content of the control. May be provided as string or as
+#'   HTML generated with Shiny/htmltools tags
+#' @param position position of control: 'topleft', 'topright',
+#'   'bottomleft', or 'bottomright'
+#' @param controlId the id of this control
+#' @param classes extra CSS classes to append to the control
+#'
+#' @describeIn map-layers Add arbitrary HTML controls to the map
+#' @export
+addControl = function(map,
+    html,
+    position=c('topleft', 'topright', 'bottomleft', 'bottomright'),
+    controlId=NULL,
+    classes=c('info', 'legend'),
+    data=getMapData(map)) {
```

Comment:
I'm not sure if the `data` argument is necessary here. Probably not. `addControl()` is not like other `add*` functions that need the `data` object to derive lat/lng and other argument values.